pandas/tests/util/test_assert_series_equal.py

```diff
@@ -281,3 +281,13 @@ class MySeries(Series):
 
     with pytest.raises(AssertionError, match="Series classes are different"):
         tm.assert_series_equal(s3, s1, check_series_type=True)
+
+
+def test_series_equal_exact_for_nonnumeric():
+    # https://github.com/pandas-dev/pandas/issues/35446
+    s1 = Series(["a", "b"])
+    s2 = Series(["a", "b"])
+    s3 = Series(["b", "a"])
+
+    _assert_series_equal_both(s1, s2, check_exact=True)
```

Comment:
can you just use tm.assert_series_equal rather than the internal function

Reply:
👍